rskj-core/src/main/java/co/rsk/config/ConfigLoader.java

```diff
@@ -55,14 +51,59 @@ public ConfigLoader(CliArgs<NodeCliOptions, NodeCliFlags> cliArgs) {
         this.cliArgs = Objects.requireNonNull(cliArgs);
     }
 
+    /**
+     * Loads configurations from different sources with the following precedence:
+     * 1. Command line arguments
+     * 2. Environment variables
+     * 3. System properties
+     * 4. User configuration file
+     * 5. Installer configuration file
+     * 6. Default settings per network in resources/[network].conf
+     * 7. Default settings for all networks in resources/reference.conf
+     *
+     * <p>
+     * If the <b><blockchain.config.verify/b> setting is {@code true} (either set in a .conf file or via <b>--verify-config</b> command line flag),
+     * then the loaded config wll be tested against the setting names defined in the expected.conf config file.
+     *
+     * Note:
+     *  1. The <b><blockchain.config.verify/b> setting is {@code false} by default.
+     *  2. Config verification process of matching actual and expected configs is recursive and takes into account appropriate setting names
+     *      and values which are collections of other settings (LIST's and OBJECT's).
+     *
+     * @throws RskConfigurationException on configuration errors
+     *
+     * @see ConfigProblems
+     */
     public Config getConfig() {
-        Config userConfig = getConfigFromCliArgs()
-                .withFallback(ConfigFactory.systemProperties())
-                .withFallback(ConfigFactory.systemEnvironment())
-                .withFallback(getUserCustomConfig())
-                .withFallback(getInstallerConfig());
+        Config cliConfig = getConfigFromCliArgs();
+        Config systemPropsConfig = ConfigFactory.systemProperties();
+        Config systemEnvConfig = ConfigFactory.systemEnvironment();
+        Config userCustomConfig = getUserCustomConfig();
+        Config installerConfig = getInstallerConfig();
+
+        Config userConfig = ConfigFactory.empty()
+                .withFallback(cliConfig)
+                .withFallback(systemPropsConfig)
+                .withFallback(systemEnvConfig)
+                .withFallback(userCustomConfig)
+                .withFallback(installerConfig);
         Config networkBaseConfig = getNetworkDefaultConfig(userConfig);
-        return userConfig.withFallback(networkBaseConfig);
+        Config unifiedConfig = userConfig.withFallback(networkBaseConfig);
+
+        if (unifiedConfig.getBoolean(SystemProperties.PROPERTY_BC_VERIFY)) {
```

Comment:
Can't we run the config verification always and just abort execution only when this flag is set?

Reply:
that would be interesting. In which case we catch the exception and only log it?

Reply:
About what exception are we talking? I only see `verify` throwing an exception if there is a malformed entry in `expected.conf`.

Reply:
sorry, I did not read the code carefully between all the changes. Right now this method throws the exception, it could skip throwing the exception if the flag is not set.